website/src/pages/docs/handlers/new-openapi.mdx

```diff
@@ -0,0 +1,221 @@
+
+### Unnamed types
+
+We use path-based naming. So names could be structured like, for example, `query_getUsers_items_firstName`
+
+## Dynamic Header Values
+
+Mesh can take dynamic values from the GraphQL Context or the environmental variables. For example, if you use `mesh dev` or `mesh start`, GraphQL Context will be the incoming HTTP request.
+
+The expression inside dynamic values should be as in JS.
+
+### From Context (HTTP Header for `mesh dev` or `mesh start`)
+
+```yaml
+sources:
+  - name: MyGraphQLApi
+    handler:
+      openapi:
+        source: ./my-schema.json
+        operationHeaders:
+          # Please do not use capital letters while getting the headers
+          Authorization: Bearer {context.headers['x-my-api-token']}
+          # You can also access to the cookies like below;
+          # Authorization: Bearer {context.cookies.myApiToken}
+```
+
+And for `mesh dev` or `mesh start`, you can pass the value using `x-my-graphql-api-token` HTTP header.
+
+### From Environmental Variable
+
+`MY_API_TOKEN` is the name of the environmental variable you have the value.
+
+```yaml {7}
+sources:
+  - name: MyGraphQLApi
+    handler:
+      openapi:
+        source: ./my-schema.json
+        operationHeaders:
+          Authorization: Bearer {env.MY_API_TOKEN}
+```
+
+## Advanced cookies handling
+
+When building a web application, cookies are often used for authentication for security reasons. On the other end, mobile applications tend to use an HTTP header.
+
+This section shows how to configure GraphQL Mesh to accept either and use GraphQL Mesh to set/unset cookies on the login & logout mutations.
+
+### Accepting one of the cookies, header, or context value
+
+We want to accept one of the following:
+
+- an `accessToken` cookie
+- an `Authorization` header
+- an authorization value available in context (e.g. set by a GraphQL auth plugin)
+
+And transmit it to the Rest API as an `Authorization` header. GraphQL Mesh does not allow dynamic selection in the `meshrc.yaml` file, but that's fine! We can use a bit of trickery.
+
+```yaml
+sources:
+  - name: Rest
+    handler:
+      openapi:
+        source: ./openapi.yaml
+        baseUrl: '{env.REST_URL}/api/'
+        operationHeaders:
+          Authorization-Header: '{context.headers.authorization}'
+          Authorization-Cookie: Bearer {context.cookies.accessToken}
+
+# You can provide a custom fetch function to override the default fetch function
+customFetch: ./src/custom-fetch.js
+```
+
+Here in the `meshrc.yaml` configuration we store the cookie in `Authorization-Cookie`, and the header in `Authorization-Header`. To introduce the logic needed to generate the proper `Authorization` header for the Rest API, we need to implement a `customFetch`. It will replace the `fetch` used by GraphQL Mesh to call the Rest API.
+
+```js
+const fetch = require('node-fetch')
+
+module.exports = (url, args, context) => {
```

Comment:
Maybe we should give an example with Envelop plugins instead? What do you think? Or do you think this is out of scope.

Reply:
Out of scope IMO.
we can note that this could be done using Envelop plugin with a link to the relevant doc